### PR TITLE
[TAS-1522] ✨ Add Airtable record for each sales

### PR DESCRIPTION
**Pull request by nwingt**

### CODE REVIEW

File: src/util/airtable.ts [hunk 1]
```diff
@@ -158,4 +169,199 @@ export async function queryAirtableForPublication({ query }) {
...
+  let paymentMethod = 'unknown';
+  let balanceTxAmount = 0;
+  let balanceTxCurrency: string | undefined;
+  let balanceTxExchangeRate: number | undefined;
+  let balanceTxNetAmount = 0;
+  let stripeFee = 0;
+  let stripeFeeCurrency = 'usd';
+  let feeTotal = 0;
+
+  if (pi.latest_charge && typeof pi.latest_charge !== 'string') {
```
Comment (williamchong) [inline src/util/airtable.ts]: Should we fetch object from stripe if this is string?

File: src/util/airtable.ts [hunk 2]
```diff
@@ -158,4 +169,199 @@ export async function queryAirtableForPublication({ query }) {
...
+    const paymentMethodDetails = pi.latest_charge?.payment_method_details;
+    if (paymentMethodDetails) {
+      paymentMethod = paymentMethodDetails.type;
+      if (paymentMethod === 'card' && paymentMethodDetails.card?.brand) {
+        paymentMethod = paymentMethodDetails.card.brand;
+      }
+    }
+
+    const balanceTx = pi.latest_charge.balance_transaction;
+    if (balanceTx && typeof balanceTx !== 'string') {
```
Comment (williamchong) [inline src/util/airtable.ts]: Should we fetch object from stripe if this is string?

File: src/util/airtable.ts [hunk 3]
```diff
@@ -158,4 +169,199 @@ export async function queryAirtableForPublication({ query }) {
...
+
+      balanceTx.fee_details.forEach(({
+        type,
+        amount,
+        currency,
+      }) => {
+        switch (type) {
+          case 'stripe_fee':
+            stripeFee = amount / 100;
+            stripeFeeCurrency = currency;
```
Comment (williamchong) [inline src/util/airtable.ts]: Should use .find instead?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/util/airtable.ts (hunk 2 above)
Comment (nwingt) [reply src/util/airtable.ts]: It shouldn't be a string as we added `expand` param in [here](https://github.com/likecoin/likecoin-api-public/blob/6e769f6a83ddf71a86df1fcc6410bf2330bcbcba/src/util/api/likernft/book/collection/purchase.ts#L461-L462)

File: src/util/airtable.ts (hunk 3 above)
Comment (nwingt) [reply src/util/airtable.ts]: Updated

File: src/util/airtable.ts (hunk 2 above)
Comment (williamchong) [reply src/util/airtable.ts]: Add a console error if we encounter string? Someone might misuse this function without knowing

**Review by williamchong — APPROVED** [review]

**Pull request merged**